tests/ui/implicit_transmute_types.rs

```diff
@@ -0,0 +1,14 @@
+//@run-rustfix
+#![allow(unused)]
+#![warn(clippy::implicit_transmute_types)]
+
+use std::mem;
+
+fn main() {
+    unsafe {
+        let _: u32 = mem::transmute(123i32);
+
+        let _: u32 = mem::transmute::<i32, u32>(123i32);
+        let _: u32 = mem::transmute::<_, _>(123i32);
```

Comment:
Can you add some more tests, with structs and other more complex types? Maybe play with `String`, `struct A(u32)` etc. These files don't get executed, so the transmutes can be unsound, without spawning nostril dragons :wink: